### PR TITLE
dolphin-emu-primehack: 1.0.5 -> 1.0.6

### PR DESCRIPTION
**Pull request by SuperSamus**
###### Description of changes
https://github.com/shiiion/dolphin/releases/tag/1.0.6

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/emulators/dolphin-emu/primehack.nix [hunk 1]
```diff
@@ -146,6 +146,6 @@ stdenv.mkDerivation rec {
     # x86_32 is an unsupported platform.
     # Enable generic build if you really want a JIT-less binary.
     broken = stdenv.isDarwin;
-    platforms = [ "x86_64-linux" "x86_64-darwin" ];
+    platforms = [ "x86_64-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/dolphin-emu/primehack.nix:149]:
```suggestion
    platforms = platforms.unix;
```

File: pkgs/applications/emulators/dolphin-emu/primehack.nix [hunk 2]
```diff
@@ -146,6 +146,6 @@ stdenv.mkDerivation rec {
     # x86_32 is an unsupported platform.
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/dolphin-emu/primehack.nix:146]:
```suggestion
```
i686 is barely supported.

Comment (SuperSamus) [reply pkgs/applications/emulators/dolphin-emu/primehack.nix:146]: I'll do this to regular Dolphin too, then.

**Comment by SuperSamus** [comment]
Wait, I messed up...

**Comment by SuperSamus** [comment]
Ok, fixed it.

**Pull request merged**